test/e2e/federated-service.go

```diff
@@ -214,6 +214,12 @@ func waitForFederatedServiceShard(cs *release_1_3.Clientset, namespace string, s
 	if numSvcs > 0 && service != nil {
 		// Renaming for clarity/readability
 		clSvc := clSvcList.Items[0]
+
+		serviceCopy, err := api.Scheme.DeepCopy(clSvc)
```

Comment:
Will be good to add a comment here saying we are removing the clusterIP before comparing because that will be different in each shard.